client/src/components/carousel.jsx

```diff
@@ -3,27 +3,75 @@ import Scaled from './scaled.jsx';
 import axios from 'axios';
 import styles from './styles.css.js';
 
-
 const dummy = [ //local drive loading for development only
   {url: './placeholder.jpg', sort: 0},
-  {url: './placeholder.jpg', sort: 1},
-  {url: './placeholder.jpg', sort: 2},
-  {url: './placeholder.jpg', sort: 3},
-  {url: './placeholder.jpg', sort: 4},
-  {url: './placeholder.jpg', sort: 5},
-  {url: './placeholder.jpg', sort: 6}
+  {url: './placeholder1.jpg', sort: 1},
+  {url: './placeholder2.jpg', sort: 2},
+  {url: './placeholder3.jpg', sort: 3},
+  {url: './placeholder4.jpg', sort: 4},
+  {url: './placeholder5.jpg', sort: 5},
+  {url: './placeholder6.jpg', sort: 6}
 ];
 
 class Carousel extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       images: dummy,
-      scaled: dummy[0]
+      scaled: 0,
+      isFavorite: false //there is no way to keep track of users right now.
     };
+
     this.retrieveImageDocument = this.retrieveImageDocument.bind(this);
+    this.changeFavorite = this.changeFavorite.bind(this);
+    this.cycleForward = this.cycleForward.bind(this);
+    this.cycleBack = this.cycleBack.bind(this);
+    this.handleThumbnailClick = this.handleThumbnailClick.bind(this);
   }
 
+  //******************** EVENT HANDLERS ********************
+  changeFavorite() {
+    this.setState({
+      isFavorite: !this.state.isFavorite
+    });
+    //TODO: need to visually show the change, for now just log
+    console.log(`Is this product favorited? ${this.state.isFavorite ? 'yes' : 'no'}`);
+  }
+
+  handleThumbnailClick(val) {
+    this.setState({
+      scaled: val
```

Comment:
Where is this val?

Reply:
Oh, I'm guessing it increments every time it's clicked

Reply:
It's the photo index that was clicked it, it comes from `thumbnail.jsx`, which is receives from the `map` function in `thumbnails.jsx`